src/framework/mod.rs

```diff
@@ -0,0 +1,24 @@
+use crate::net::MpcServerError;
+use axum::response::{IntoResponse, Response};
+use thiserror::Error;
+pub mod step;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("invalid step")]
```

Comment:
it is a good practice to include the invalid value into error description (unless it is a secret of course)

Reply:
i agree. i'll think about how best to represent that

Reply:
I think it could be enough to say this

```rust
pub enum Error {
    #[error("invalid step: {0}")]
    InvalidStep(String)
```